core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -848,15 +855,81 @@ public Graph build() {
+                LOGGER.info("No FunctionAuthoriser hook was supplied, adding default hook.");
+                hooks.add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+            }
+        }
+
+        private void updateNamedViewResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedView.class,
+                    NamedViewResolver.class,
+                    null != properties ? properties.getCacheServiceNamedViewSuffix(config.getGraphId()) : null);
+        }
+
+        private void updateNamedOperationResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedOperation.class,
+                    NamedOperationResolver.class,
+                    null != properties ? properties.getCacheServiceNamedOperationSuffix(config.getGraphId()) : null);
+        }
+
+        /**
+         * This method is designed to keep suffix used for cache names aligned between
+         * a Handler which Adds to the cache and ResolverHook which Gets from the cache.
+         * If the suffix is mismatched then writing and reading to cache will not behave correctly.
+         *
+         * @param config               the graphConfig containing the Hooks
+         * @param operationClass       the Operation requiring cache write
+         * @param hookClass            the Hook requiring cache reading
+         * @param suffixFromProperties the suffix from property
+         */
+        private void updateGetFromCacheHookWhenMissingOrHasWrongSuffix(final GraphConfig config, final Class<? extends Operation> operationClass, final Class<? extends GetFromCacheHook> hookClass, final String suffixFromProperties) {
+            if (store.isSupported(operationClass)) {
+                //Get Handler
+                final OperationHandler<Operation> addToCacheHandler = store.getOperationHandler(operationClass);
+                //If Handler is a AddToCacheHandler
+                final String suffix;
+                if (AddToCacheHandler.class.isAssignableFrom(addToCacheHandler.getClass())) {
+                    // get Suffix
+                    suffix = ((AddToCacheHandler<?>) addToCacheHandler).getSuffixCacheName();
+                } else {
+                    // otherwise get from properties
+                    LOGGER.warn(String.format(HANDLER_WAS_NOT_EXPECTED_TYPE_ADD_TO_CACHE_HANDLER, operationClass, AddToCacheHandler.class.getSimpleName(), suffixFromProperties));
+                    suffix = suffixFromProperties;
+                }
+                final List<GraphHook> hooks = config.getHooks();
+                //Is GetFromCacheHook missing
+                if (!hasHook(hooks, hookClass)) {
+                    //provide info about the not having required.
```

Comment:
Is this what was meant?
```suggestion
                    //provide info about the graph not having the required hook
```